h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderHelper.java

```diff
@@ -51,14 +59,26 @@ public static int addKFoldColumn(Frame frame, String name, int nfolds, long seed
     return frame.numCols() - 1;
   }
 
-  static double calculatePriorMean(Frame encodings) {
-    Vec numeratorVec = encodings.vec(NUMERATOR_COL);
-    Vec denominatorVec = encodings.vec(DENOMINATOR_COL);
-    assert numeratorVec != null;
-    assert denominatorVec != null;
-    return numeratorVec.mean() / denominatorVec.mean();
+  static double computePriorMean(Frame encodings) {
+    assert encodings.find(TARGETCLASS_COL) < 0;
+    return computePriorMean(encodings, NO_TARGET_CLASS);
+  }
+  
+  static double computePriorMean(Frame encodings, int targetClass) {
+    int tcIdx = encodings.find(TARGETCLASS_COL);
+    assert (targetClass == NO_TARGET_CLASS) == (tcIdx < 0);
+    Frame fr = null;
```

Comment:
Note to self: scope would work as well

Reply:
agree, there's only one frame to manage though, so the direct call to delete may be simpler in this case.